appengine/cloudsql/createTables.js

```diff
@@ -13,71 +13,127 @@
+    return knex;
+  });
 }
 // [END createTable]
 
-// [START getConnection]
-const FIELDS = ['user', 'password', 'database'];
-
+// [START postgresGetConnection]
 /**
  * Ask the user for connection configuration and create a new connection.
- *
- * @param {function} callback The callback function.
  */
-function getConnection (callback) {
-  prompt.start();
-  prompt.get(FIELDS, (err, config) => {
-    if (err) {
-      callback(err);
-      return;
-    }
+function postgresGetConnection () {
+  const FIELDS = ['user', 'password', 'database'];
+  return new Promise((resolve, reject) => {
+    prompt.start();
+    prompt.get(FIELDS, (err, config) => {
+      if (err) {
+        reject(err);
+      }
 
-    const user = encodeURIComponent(config.user);
-    const password = encodeURIComponent(config.password);
-    const database = encodeURIComponent(config.database);
-
-    const uri = `mysql://${user}:${password}@127.0.0.1:3306/${database}`;
-    callback(null, mysql.createConnection(uri));
+      // Connect to the database
+      resolve(Knex({
+        client: 'pg',
+        connection: config
+      }));
+    });
   });
 }
-// [END getConnection]
+// [END postgresGetConnection]
+
+// [START mysqlGetConnection]
+/**
+ * Ask the user for connection configuration and create a new connection.
+ */
+function mysqlGetConnection () {
+  const FIELDS = ['user', 'password', 'database'];
+  return new Promise((resolve, reject) => {
+    prompt.start();
+    prompt.get(FIELDS, (err, config) => {
+      if (err) {
+        return reject(err);
+      }
 
-// [START main]
-getConnection((err, connection) => {
-  if (err) {
-    console.error(err);
-    return;
-  }
-  createTable(connection, (err, result) => {
-    connection.end();
-    if (err) {
-      console.error(err);
-      return;
-    }
-    console.log(result);
+      // Connect to the database
+      return resolve(Knex({
+        client: 'mysql',
+        connection: config
+      }));
+    });
   });
-});
-// [END main]
+}
+// [END mysqlGetConnection]
+
+exports.postgresMain = function () {
+  // [START postgresMain]
+  postgresGetConnection()
+    .then((knex) => {
+      return createTable(knex);
+    })
+    .then((knex) => {
+      return knex.destroy();
+    })
+    .catch((err, knex) => {
+      console.error(`Failed to create database connection:`, err);
+      return knex.destroy();
+    });
+  // [END postgresMain]
+};
+
+exports.mysqlMain = function () {
+  // [START mysqlMain]
+  mysqlGetConnection()
+    .then((knex) => {
+      return createTable(knex);
+    })
+    .then((knex) => {
+      return knex.destroy();
+    })
+    .catch((err, knex) => {
+      console.error(`Failed to create database connection:`, err);
+      if (knex) {
+        knex.destroy();
+      }
+    });
+  // [END mysqlMain]
+};
 // [END createTables]
+
+// Get type of SQL client to use
+const sqlClient = process.env.SQL_CLIENT;
+if (sqlClient === 'pg') {
```

Comment:
no need for conditional if we just rely on passing `process.env.SQL_CLIENT` to the Knex constructor